src/SignalRServiceExtension/TriggerBindings/SignalRTriggerBindingProvider.cs

```diff
@@ -33,8 +39,68 @@ public Task<ITriggerBinding> TryCreateAsync(TriggerBindingProviderContext contex
             {
                 return Task.FromResult<ITriggerBinding>(null);
             }
+            var resolvedAttribute = GetParameterResolvedAttribute(attribute);
+            ValidateSignalRTriggerAttributeBinding(resolvedAttribute);
+            
+            return Task.FromResult<ITriggerBinding>(new SignalRTriggerBinding(parameterInfo, resolvedAttribute, _dispatcher));
+        }
+
+        private SignalRTriggerAttribute GetParameterResolvedAttribute(SignalRTriggerAttribute attribute)
+        {
+            //TODO: AutoResolve more properties in attribute
+            var resolvedConnectionString = GetResolvedConnectionString(
+                typeof(SignalRTriggerAttribute).GetProperty(nameof(attribute.ConnectionStringSetting)),
+                attribute.ConnectionStringSetting);
+
+            return new SignalRTriggerAttribute(attribute.HubName, attribute.Category, attribute.Event, attribute.ParameterNames){ConnectionStringSetting = resolvedConnectionString};
+        }
 
-            return Task.FromResult<ITriggerBinding>(new SignalRTriggerBinding(parameterInfo, attribute, dispatcher));
+        private string GetResolvedConnectionString(PropertyInfo property, string configurationName)
+        {
+            string resolvedConnectionString;
+            if (!string.IsNullOrWhiteSpace(configurationName))
+            {
+                resolvedConnectionString = _nameResolver.Resolve(configurationName);
+            }
+            else
+            {
+                var attribute = property.GetCustomAttribute<AppSettingAttribute>();
+                if (attribute == null)
+                {
+                    throw new InvalidOperationException($"Unable to get AppSettingAttribute on property {property.Name}");
+                }
+                resolvedConnectionString = _nameResolver.Resolve(attribute.Default);
+            }
+
+            if (string.IsNullOrEmpty(resolvedConnectionString))
+            {
+                resolvedConnectionString = _options.ConnectionString;
+            }
+
+            return resolvedConnectionString;
```

Comment:
return string.IsNullOrEmpty(resolvedConnectionString) ? _options.ConnectionString : resolvedConnectionString;

Reply:
use the Utils.FirstOrDefault?

Reply:
But there're only two parameters, I think it's simple and efficient enough to use `?:`